edgelet/edgelet-docker/src/image_prune_data.rs

```diff
@@ -0,0 +1,646 @@
+                .into_string()
+                .unwrap(),
+            test_file_dir
+                .join(IMAGE_USE_FILENAME)
+                .into_os_string()
+                .into_string()
+                .unwrap(),
+        );
+
+        let curr_time: String = format!("{}{}", Utc::now().hour(), Utc::now().minute());
+        let mut settings = ImagePruneSettings::new(
+            Duration::from_secs(30),
+            Duration::from_secs(5),
+            curr_time,
+            false,
+        );
+        let mut image_use_data = ImagePruneData::new(&test_file_dir, Some(settings)).unwrap();
+
+        let mut in_use_image_ids: HashSet<String> = HashSet::new();
+        in_use_image_ids.insert(
+            "sha256:670dcc86b69df89a9d5a9e1a7ae5b8f67619c1c74e19de8a35f57d6c06505fd4".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:7a45202c8491b92b7e7a9a0cbf887079fcdb86ea3cb0b7a4cb8f5491281e985d".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:a40d3130a63918663f6e412178d2e83010994bb5a6bdb9ba314ca43013c05331".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:85fdb1e9675c837c18b75f103be6f156587d1058eced1fc508cdb84a722e4f82".to_string(),
+        );
+
+        unsafe { sleep(6) };
+
+        // image prune disable... don't remove stuff
+        let mut images_to_delete = image_use_data
+            .prune_images_from_file(in_use_image_ids.clone())
+            .unwrap();
+        assert!(images_to_delete.is_empty());
+
+        // image prune enabled, remove stuff
+        let curr_time: String = format!("{}{}", Utc::now().hour(), Utc::now().minute());
+        settings = ImagePruneSettings::new(
+            Duration::from_secs(30),
+            Duration::from_secs(5),
+            curr_time,
+            true,
+        );
+        image_use_data = ImagePruneData::new(&test_file_dir, Some(settings)).unwrap();
+
+        images_to_delete = image_use_data
+            .prune_images_from_file(in_use_image_ids)
+            .unwrap();
+        assert!(images_to_delete.len() == 2);
+
+        // cleanup
+        std::fs::remove_dir_all(test_file_dir).unwrap();
+    }
+
+    /* =============================================================== MORE TESTS ============================================================ */
+
+    #[test]
+    #[serial]
+    #[should_panic]
+    fn test_panic_write_images_with_timestamp() {
+        let mut result = write_images_with_timestamp(
+            &HashMap::new(),
+            "/etc/other_file".to_string(),
+            IMAGE_USE_FILENAME.to_string(),
+        );
+        assert!(result.is_err());
+
+        result = write_images_with_timestamp(
+            &HashMap::new(),
+            TMP_FILENAME.to_string(),
+            "/etc/other_file".to_string(),
+        );
+        assert!(result.is_err());
+    }
+
+    #[test]
+    #[serial]
+    fn test_file_rename_write_images_with_timestamp() {
```

Comment:
It's not clear to me what the intent of this test is RE 'file rename'. I see that it checks that creating the image use file succeeds when an empty `HashMap` is passed into `write_images_with_timestamp()`? Is there something else I'm missing? A comment and/or different test name would help with clarity.

Reply:
done. something simple to test code that otherwise would remain not covered with unit test